src/physics/em/detail/Utils.hh

```diff
@@ -0,0 +1,55 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file Utils.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <unordered_map>
+#include "base/Macros.hh"
+#include "base/Types.hh"
+#include "physics/em/AtomicRelaxationInterface.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Helper class for calculating the maximum possible number of secondaries
+ * produced in atomic relaxation for a given element and electron/photon
+ * production threshold.
+ */
+class MaxSecondariesCalculator
+{
+  public:
+    //!@{
+    //! Type aliases
+    using MevEnergy = units::MevEnergy;
+    //!@}
+
+  public:
+    // Construct with EADL transition data and production thresholds
+    MaxSecondariesCalculator(const AtomicRelaxElement& el,
+                             MevEnergy                 electron_cut,
+                             MevEnergy                 gamma_cut);
+
+    // Calculate the maximum possible number of secondaries produced
+    size_type operator()();
+
+  private:
+    Span<const AtomicRelaxSubshell>           shells_;
+    const real_type                           electron_cut_;
+    const real_type                           gamma_cut_;
+    std::unordered_map<SubshellId, size_type> visited_;
+
+    // HELPER FUNCTIONS
+
+    size_type calc(SubshellId vacancy_shell, size_type count);
+};
```

Comment:
Since this class doesn't have any function arguments and it has a persistent `visted_` state that will cause multiple calls to give different answers, this class might be better as an implementation detail of a free `calc_max_secondaries` function.